### PR TITLE
Closes #3996: Remove workarounds for opening custom tab in browser.

### LINKED ISSUE #3996: Remove workarounds for opening custom tab in browser
We now allow releasing a session from an `EngineView` and no longer need workarounds like delaying opening the browser or removing the view.

### PR DESCRIPTION
**Pull request by pocmo**
`EngineView` now allows releasing the `Session` so that it can be rendered by an
other `EngineView` instance.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by pocmo** [comment]
@ekager I did test switching from a custom tab to the browser. Are there any other edge cases I should test?

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -769,15 +764,24 @@ class BrowserFragment : Fragment(), BackHandler {
...
                 }
+
+                // Close this activity since it is no longer displaying any session
                 activity?.finish()
+
+                // Now switch to the actual browser which should now display our new selected session
+                val intent = Intent(context, IntentReceiverActivity::class.java).also {
+                    it.action = Intent.ACTION_VIEW
+                    it.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                }
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: I was a bit surprised that we go to `IntentReceiverActivity` and not to `BrowserActivity` directly. But I also didn't feel adventurous enough to try. :)

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 2]
```diff
@@ -769,15 +764,24 @@ class BrowserFragment : Fragment(), BackHandler {
...
                     requireComponents.core.sessionManager.select(it)
                 }
+
+                // Close this activity since it is no longer displaying any session
                 activity?.finish()
+
+                // Now switch to the actual browser which should now display our new selected session
+                val intent = Intent(context, IntentReceiverActivity::class.java).also {
+                    it.action = Intent.ACTION_VIEW
+                    it.flags = Intent.FLAG_ACTIVITY_NEW_TASK
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Added `NEW_TASK` here so that the original app task remains available to switch back to the other app - otherwise the browser may take that over.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: We don't have a BrowserActivity 😅

**Comment by ekager** [comment]
The pendingOpenInBrowser intent was because of a weird lifecycle issue: https://github.com/mozilla-mobile/fenix/issues/3739
Did you test-
1. Open custom tab
2. Open in Fenix from menu
3. Three dot menu -> Use FIP 

I'm not sure I see how this would fix that issue unless there's some more magic on the AC session selection side or NEW_TASK would fix it :P

**Comment by pocmo** [comment]
> Did you test-

No! I'll try that now. I thought this was also caused by the `Session` not being released. I'll test.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Okay, `HomeActivity`? 🙃

**Comment by pocmo** [comment]
Oh, yeah, I re-introduced this bug.

This is caused by `FindInPageIntegration.launch` (which is static) being set from `start()` and `stop()` - and unfortunately the new activity is in started state before the other one is in stopped state. Which means we execute `stop()` launch which **globally** clears `FindInPageIntegration.launch` causing FIP to never launch.

I think we may be able to fix that differently (by keeping a `launch` reference per lifecycle/fragment). But that's out of scope for this PR and worth a new issue. I'll add the pendingIntent back in for now. :)

**Comment by pocmo** [comment]
@ekager Updated the PR. The pending intent is back in for now. :)
